relayer/src/foreign_client.rs

```diff
@@ -1144,6 +1167,10 @@ impl<DstChain: ChainHandle, SrcChain: ChainHandle> ForeignClient<DstChain, SrcCh
             // Check for misbehaviour according to the specific source chain type.
             // In case of Tendermint client, this will also check the BFT time violation if
             // a header for the event height cannot be retrieved from the witness.
+            //
+            // FIXME: This returns error if the update event contains expired client state.
+            // This can happen even if the latest client state is unexpired, but the
+            // update event is from earlier.
```

Comment:
What should we do about this? Leave for future PR?

Reply:
I have tried to reproduce the error with a simpler test and added it here as a manual test. I also made the misbehavior task always continue after initial check, so that it can still act on new client update events.

Reply:
Looks good to me! I guess we can  now remove the FIXME